net/security-context/tls-stream.cpp

```diff
@@ -339,6 +340,11 @@ class TLSSocketStream : public ForwardSocketStream {
         SSL_free(ssl);
     }
 
+    bool set_hostname(const char* hostname) {
+        int ret = SSL_set_tlsext_host_name(ssl, hostname);
```

Comment:
`SSL_set_tlsext_host_name` is not provided in OpenSSL 1.0.2 series API. The code here still should considering compatibility to OpenSSL 1.0.2.

it is better to use macro check `OPENSSL_VERSION_NUMBER` and use `SSL_ctrl` to set host name when version lower than 1.1.0.